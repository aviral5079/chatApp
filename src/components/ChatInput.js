// File: ChatInput.js
import React from "react";

const ChatInput = ({ value, setValue, getMessages }) => {
  return (
    <div className="bottom-section">
      <div className="input-container">
        <input
          data-testid="input"
          className="text-box"
          value={value}
          onChange={(e) => setValue(e.target.value)}
        />
        <div data-testid="submit" id="submit" onClick={getMessages}>
          â†µ
        </div>
      </div>
      <p className="info"></p>
    </div>
  );
};

export default ChatInput;
