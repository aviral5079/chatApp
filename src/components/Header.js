import React from "react";
import PropTypes from "prop-types";

const Header = ({ text }) => {
  return (
    <div data-testid="text">
      <h1>{text || "Assistant"}</h1>
    </div>
  );
};

// Header.propTypes = {
//   text: PropTypes.string,
// };

export default Header;
