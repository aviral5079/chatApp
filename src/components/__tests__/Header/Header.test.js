import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import ChatInput from "../../ChatInput";
import Header from "../../Header";
import "@testing-library/jest-dom";

afterEach(() => {
  cleanup();
});

test("should render header component", () => {
  render(<Header />);

  var textElem = screen.getByTestId("text");

  expect(textElem).toBeInTheDocument();
});

test("text matches snapshot", () => {
  const component = renderer.create(<Header />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("header text changes after clicking submit button", () => {
  const { getByTestId } = render(<Header />);
  const { getByTestId: getByTestId2 } = render(<ChatInput />);

  fireEvent.change(getByTestId2("input"), {
    target: { value: "How are you today?" },
  });
  fireEvent.click(getByTestId2("submit"));

  expect(getByTestId("text")).toHaveTextContent("How are you today?");
});
