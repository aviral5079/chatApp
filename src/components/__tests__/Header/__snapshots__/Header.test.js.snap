// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text matches snapshot 1`] = `
<div
  data-testid="text"
>
  <h1>
    Assistant
  </h1>
</div>
`;
